services/datamanager/builtin/builtin.go

```diff
@@ -679,3 +712,19 @@ func (svc *builtIn) updateDataCaptureConfigs(
 func generateMetadataKey(component, method string) string {
 	return fmt.Sprintf("%s/%s", component, method)
 }
+
+//nolint:unparam
+func pollFilesystem(ctx context.Context, wg *sync.WaitGroup, captureDir string, syncer *datasync.Manager, logger logging.Logger) {
+	for {
+		t := time.NewTimer(filesystemPollInterval)
+		select {
+		case <-ctx.Done():
+			t.Stop()
+			logger.Debug("Exiting fs polling thread")
+			wg.Done()
+			return
+		case <-t.C:
+			logger.Debug("Polling")
```

Comment:
I know this is debugging for now, but reminder to remove before merging. Then useful to re-add when we have actual logic here.